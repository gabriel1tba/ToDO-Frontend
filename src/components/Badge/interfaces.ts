export interface IBadge {
  title: string | number;
  fontColor: string;
  backgroundColor: string;
}
