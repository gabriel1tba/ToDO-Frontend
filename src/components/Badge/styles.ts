import styled, { css } from 'styled-components';

import { IBadge } from '.';

export const Wrapper = styled.span<Omit<IBadge, 'title'>>`
  ${({ color, background }) => css`
    cursor: default;

    display: inline-block;
    border-radius: 10px;
    text-align: start;
    font-weight: 700;
    font-size: 14px;
    line-height: 15px;
    height: 25px;
    color: ${color};
    background: ${background};

    padding: 0.3rem 0.6rem 0.1rem;
  `}
`;
