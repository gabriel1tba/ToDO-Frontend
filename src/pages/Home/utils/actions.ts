import { TTodo } from 'services/TodoService/interfaces';

export enum Types {
  GET_TODOS = 'GET_TODOS',
  CREATE_TODO = 'CREATE_TODO',
  UPDATE_TODO = 'UPDATE_TODO',
  DELETE_TODO = 'DELETE_TODO',
}

interface GetTodos {
  type: Types.GET_TODOS;
  payload: TTodo[];
}

interface CreateTodo {
  type: Types.CREATE_TODO;
  payload: TTodo;
}

interface UpdateTodo {
  type: Types.UPDATE_TODO;
  payload: TTodo;
}

interface DeleteTodo {
  type: Types.DELETE_TODO;
  payload: string;
}

export type Action = GetTodos | CreateTodo | UpdateTodo | DeleteTodo;

export const getTodosAction = (todos: TTodo[]): GetTodos => ({
  type: Types.GET_TODOS,
  payload: todos,
});

export const createTodoAction = (todo: TTodo): CreateTodo => ({
  type: Types.CREATE_TODO,
  payload: todo,
});

export const updateTodoAction = (todo: TTodo): UpdateTodo => ({
  type: Types.UPDATE_TODO,
  payload: todo,
});

export const deleteTodoAction = (id: string): DeleteTodo => ({
  type: Types.DELETE_TODO,
  payload: id,
});
