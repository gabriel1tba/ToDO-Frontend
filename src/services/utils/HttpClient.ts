import axios from 'axios';

const HttpClient = axios.create({
  baseURL: `${process.env.REACT_APP_API_HOST}:${process.env.REACT_APP_API_PORT}`,
});

export default HttpClient;
