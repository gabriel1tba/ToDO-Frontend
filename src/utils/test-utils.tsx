import { render, RenderOptions } from '@testing-library/react';
import { TodoProvider } from 'context/todos';

interface IWrapperProvider {
  children: React.ReactNode;
}

const WrapperProvider = ({ children }: IWrapperProvider) => (
  <TodoProvider>{children}</TodoProvider>
);

const customRender = (ui: React.ReactElement, options?: RenderOptions) =>
  render(<WrapperProvider>{ui}</WrapperProvider>, { ...options });

export * from '@testing-library/react';
export { customRender as render };
