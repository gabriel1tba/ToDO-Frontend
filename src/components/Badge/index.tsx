import * as S from './styles';

export interface IBadge {
  title: string | number;
  color: string;
  background: string;
}

const Badge = ({ title, color, background }: IBadge) => {
  return (
    <S.Wrapper color={color} background={background}>
      {title}
    </S.Wrapper>
  );
};

export default Badge;
