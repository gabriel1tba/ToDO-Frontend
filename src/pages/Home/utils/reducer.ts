import { TTodo } from 'services/TodoService/interfaces';

import { Types, Action } from './actions';

const todosReducer = (state: TTodo[], action: Action): TTodo[] => {
  switch (action.type) {
    case Types.GET_TODOS: {
      return action.payload;
    }

    case Types.CREATE_TODO: {
      return [...state, action.payload];
    }

    case Types.UPDATE_TODO: {
      return state.map((todo) => {
        if (action.payload.id === todo.id) {
          todo = { ...action.payload, created_at: todo.created_at };
        }
        return todo;
      });
    }

    case Types.DELETE_TODO: {
      return state.filter((todo) => {
        return todo.id !== action.payload;
      });
    }

    default:
      return state;
  }
};

export default todosReducer;
