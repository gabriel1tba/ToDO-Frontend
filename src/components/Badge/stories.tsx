import { Story, Meta } from '@storybook/react';

import Badge, { IBadge } from '.';

export default {
  title: 'Badge',
  component: Badge,
  argTypes: {
    title: {
      type: 'string',
    },
    color: {
      type: 'string',
    },
    background: {
      type: 'string',
    },
  },
} as Meta;

export const Default: Story<IBadge> = (args) => <Badge {...args} />;

Default.args = {
  title: '1 de 5',
  color: '#D9D9D9',
  background: '#333333',
};

export const Success: Story<IBadge> = (args) => <Badge {...args} />;

Success.args = {
  title: 'Success',
  color: '#1e9250',
  background: '#e0fded',
};

export const Error: Story<IBadge> = (args) => <Badge {...args} />;

Error.args = {
  title: 'Error',
  color: '#FC5050',
  background: '#ffe3e3',
};
