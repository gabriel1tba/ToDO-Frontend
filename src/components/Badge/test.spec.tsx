import { render, screen } from 'utils/test-utils';

import Badge from '.';

describe('<Badge />', () => {
  it('should render the Badge correctly', () => {
    const { container } = render(
      <Badge
        title="Tarefas atuais"
        fontColor="#3498db"
        backgroundColor="rgba(52, 152, 219, 0.2)"
      />,
    );

    expect(screen.getByText(/tarefas atuais/i)).toBeInTheDocument();
    expect(container.firstChild).toHaveStyle({
      color: '#3498db',
      backgroundColor: 'rgba(52, 152, 219, 0.2)',
    });

    expect(container.firstChild).toMatchSnapshot();
  });
});
