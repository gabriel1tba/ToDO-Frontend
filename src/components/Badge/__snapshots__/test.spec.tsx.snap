// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Badge /> should render the Badge correctly 1`] = `
.c0 {
  cursor: default;
  display: inline-block;
  border-radius: 10px;
  text-align: start;
  font-weight: 700;
  font-size: 14px;
  line-height: 15px;
  height: 25px;
  color: #3498db;
  background: rgba(52,152,219,0.2);
  padding: 0.3rem 0.6rem 0.1rem;
}

<span
  class="c0"
  color="#3498db"
>
  Tarefas atuais
</span>
`;
